### PR TITLE
Fei2022 28 sort auctions

### PR DESCRIPTION
**Pull request by dorotanowak00**
Added route controller method in BE for handling sorting and filtering auctions. 
Added filtering and sorting in FE.

### CODE REVIEW

File: frontend/src/components/categoryButtons/CategoryButtons.tsx [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+    category: string;
+    time: string;
+    sortBy: string;
+}
+
+interface CategoryButtonsProps {
+    onCategorySelect: (e: React.MouseEvent) => void;
+    formState: FormState;
+}
+
```
Comment (marpuz) [inline frontend/src/components/categoryButtons/CategoryButtons.tsx:9]: suggestion: You can put this interfaces to other file

File: frontend/src/components/categoryButtons/CategoryButtons.tsx [hunk 2]
```diff
@@ -14,26 +28,28 @@ export const CategoryButtons = () => {
     interface FilterCategory {
         name: FilterCategoryType;
         label: string;
+        id: string;
     }
```
Comment (marpuz) [inline frontend/src/components/categoryButtons/CategoryButtons.tsx:28]: same here

File: frontend/src/components/selectFilters/SelectFilters.tsx [hunk 3]
```diff
@@ -1,86 +1,145 @@
...
+    category: string;
+    time: string;
+    sortBy: string;
+    ascending: string;
+}
+
+interface SelectFiltersProps {
+    onFieldSelect: (e: React.ChangeEvent) => void;
+    formState: FormState;
+}
```
Comment (marpuz) [inline frontend/src/components/selectFilters/SelectFilters.tsx:8]: and here

File: frontend/src/components/selectFilters/SelectFilters.tsx [hunk 4]
```diff
@@ -1,86 +1,145 @@
...
+        filterBy?: string;
+        ascending?: string;
     }
 
     interface FilterSelect {
         name: FilterSelectType;
         label: string;
+        defaultValue: string;
         values: FilterSelectValues[];
     }
```
Comment (marpuz) [inline frontend/src/components/selectFilters/SelectFilters.tsx:44]: and here

**Review by marpuz — APPROVED** [review]

**Review by kkolasinski — APPROVED** [review]

**Pull request merged**